src/node_http_parser.cc

```diff
@@ -187,7 +187,182 @@ struct StringPtr {
+    }
+
+    static void All(const FunctionCallbackInfo<Value>& args) {
+      v8::Isolate* isolate = args.GetIsolate();
+      Local<Context> context = isolate->GetCurrentContext();
+
+      Local<Array> expired = Array::New(isolate);
+      ConnectionsList* list;
+
+      ASSIGN_OR_RETURN_UNWRAP(&list, args.Holder());
+
+      uint32_t i = 0;
+      for (auto parser : list->allConnections) {
+        expired->Set(context, i++, parser->AsJSObject()).Check();
+      }
+
+      return args.GetReturnValue().Set(expired);
+    }
+
+    static void Idle(const FunctionCallbackInfo<Value>& args) {
+      v8::Isolate* isolate = args.GetIsolate();
+      Local<Context> context = isolate->GetCurrentContext();
+
+      Local<Array> expired = Array::New(isolate);
+      ConnectionsList* list;
+
+      ASSIGN_OR_RETURN_UNWRAP(&list, args.Holder());
+
+      uint32_t i = 0;
+      for (auto parser : list->allConnections) {
+        if (parser->last_message_start_ == 0) {
+          expired->Set(context, i++, parser->AsJSObject()).Check();
+        }
+      }
+
+      return args.GetReturnValue().Set(expired);
+    }
+
+    static void Active(const FunctionCallbackInfo<Value>& args) {
+      v8::Isolate* isolate = args.GetIsolate();
+      Local<Context> context = isolate->GetCurrentContext();
+
+      Local<Array> expired = Array::New(isolate);
+      ConnectionsList* list;
+
+      ASSIGN_OR_RETURN_UNWRAP(&list, args.Holder());
+
+      uint32_t i = 0;
+      for (auto parser : list->activeConnections) {
+        expired->Set(context, i++, parser->AsJSObject()).Check();
+      }
+
+      return args.GetReturnValue().Set(expired);
+    }
+
+    static void Expired(const FunctionCallbackInfo<Value>& args) {
+      v8::Isolate* isolate = args.GetIsolate();
+      Local<Context> context = isolate->GetCurrentContext();
+
+      Local<Array> expired = Array::New(isolate);
+      ConnectionsList* list;
+
+      ASSIGN_OR_RETURN_UNWRAP(&list, args.Holder());
+      CHECK(args[0]->IsNumber());
+      CHECK(args[1]->IsNumber());
+      uint64_t headers_timeout = args[0].As<v8::Uint32>()->Value() * 1000000;
+      uint64_t request_timeout = args[1].As<v8::Uint32>()->Value() * 1000000;
+
+      if (headers_timeout == 0 && request_timeout == 0) {
+        return args.GetReturnValue().Set(expired);
+      } else if (request_timeout > 0 && headers_timeout > request_timeout) {
+        std::swap(headers_timeout, request_timeout);
+      }
+
+      const uint64_t now = uv_hrtime();
+      const uint64_t headers_deadline =
+        headers_timeout > 0 ? now - headers_timeout : 0;
+      const uint64_t request_deadline =
+        request_timeout > 0 ? now - request_timeout : 0;
+
+      uint32_t i = 0;
+      auto iter = list->activeConnections.begin();
+      auto end = list->activeConnections.end();
+      while (iter != end) {
+        ExpirableParser* parser = *iter;
+        iter++;
+
+        // Check for expiration
```

Comment:
```suggestion
        // Check for expiration.
```
The C++ style guide wants us to use full sentences - https://github.com/nodejs/node/blob/master/doc/contributing/cpp-style-guide.md#c-style-comments.

Reply:
👍